cpp/src/table/table_device_view.cu

```diff
@@ -40,18 +41,29 @@ table_device_view_base<ColumnDeviceView, HostTableView>::table_device_view_base(
       _num_columns{source_view.num_columns()},
       _stream{stream} {
   if (source_view.num_columns() > 0) {
-    size_type total_descendants =
+    //
+    size_type views_size_bytes =
         std::accumulate(source_view.begin(), source_view.end(), 0,
-                        [](size_type init, column_view col) {
-                          return init + count_descendants(col);
-                        });
-    CUDF_EXPECTS(0 == total_descendants,
-                 "Columns with descendants are not yet supported.");
-
-    auto views_size_bytes =
-        source_view.num_columns() * sizeof(*source_view.begin());
+            [](size_type init, column_view col) {
+                return init + ColumnDeviceView::extent(col);
+            });
+    
+    std::vector<int8_t> h_buffer(views_size_bytes);
```

Comment:
I think this would be easier to understand with some documentation. Perhaps move this logic into a function in an anonymous namespace and document it there. Something like `copy_views_to_device`?

Reply:
Added more detailed comments.